File: frontEnd/src/types/domain/api/dto/AddApiInterfaceDTO.ts
import {RequestMethod} from "@/types/domain/api/apiEnum/RequestMethod";
import {RequestParameter} from "@/types/domain/api/requestModel/RequestParameter";
import {Environment} from "@/types/domain/api/Environment";
import {ApiType} from "@/types/domain/api/apiEnum/ApiType";
import {AssertParameter} from "@/types/domain/api/requestModel/AssertParameter";
import {ApiExtraction} from "@/types/domain/api/requestModel/ApiExtraction";
import {useProjectStore} from "@/store";
import useTeamStore from "@/store/modules/team";
import {Body} from "@/types/domain/api/requestModel/Body";
import {ApiNodeType} from "@/types/domain/api/apiEnum/ApiNodeType";
import { RequestExecuteInfo } from '@/types/domain/api/requestModel/RequestExecuteInfo';
import {ScriptItem} from "@/types/domain/api/requestModel/ScriptItem";
import {MockResponse} from "@/types/domain/api/requestModel/MockResponse";
import {ApiResponseExample} from "@/types/domain/api/requestModel/ApiResponseExample";
import {SqlConfig} from "@/types/domain/api/requestModel/SqlConfig";
import {AuthConfig} from "@/types/domain/api/requestModel/AuthConfig";
import {ResponseSchema} from "@/types/domain/api/requestModel/ResponseSchema";

export class AddApiInterfaceDTO {
  id?: number;
  parentId?: number;
  /**
   * 所属项目id
   */
  projectId?: number;

  /**
   * 所属团队id
   */
  teamId?: number;

  /**
   * 接口名称
   */
  apiName?: string;
  /**
   * 节点类型
   */
  apiNode?: ApiNodeType;
  /**
   * 请求方式
   */
  requestMethod?: RequestMethod;

  /**
   * 请求路径
   */
  requestPath?: string;

  /**
   * 请求头
   */
  requestHeader?: RequestParameter[];

  /**
   * cookie
   */
  cookies?: RequestParameter[];

  /**
   * query参数
   */
  query?: RequestParameter[];

  /**
   * body请求体
   */
  body?: Body;

  /**
   * 关联的环境数据
   */
  envInfo?: RequestExecuteInfo;

  /**
   * 接口类型（HTTP / SQL / TCP / WEBSOCKET）
   */
  apiType?: ApiType;

  /**
   * SQL 调试配置（apiType=SQL 时使用）
   */
  sqlConfig?: SqlConfig;

  /**
   * 排序
   */
  sort?: number;

  /**
   * 关联提取
   */
  associationExtraction?: ApiExtraction[];

  /**
   * 前置脚本列表（按 sort 排序后依次执行）
   */
  preScript?: ScriptItem[];

  /**
   * 后置脚本列表（按 sort 排序后依次执行）
   */
  postScript?: ScriptItem[];

  /**
   * Mock 响应配置
   */
  mockResponse?: MockResponse;

  /**
   * 响应定义（响应结构/数据模型，用于执行时校验）
   */
  responseSchema?: ResponseSchema;

  /**
   * 接口鉴权配置（HTTP 接口调试用）
   */
  authConfig?: AuthConfig;

  /**
   * Swagger / OpenAPI 响应示例列表
   */
  responseExamples?: ApiResponseExample[];

  /**
   * 断言
   */
  apiResultAssert?: AssertParameter[];

  /**
   * 来源id
   */
  sourceDratId?: number;

  constructor(apiNode: ApiNodeType | undefined) {
    if (apiNode === ApiNodeType.FOLDER) {
      this.parentId = 0;
      (this.projectId = useProjectStore().getProjectId ?? undefined),
        (this.teamId = useTeamStore().getTeamId ?? undefined),
        (this.apiName = ''),
        (this.apiNode = ApiNodeType.FOLDER),
        (this.requestMethod = undefined),
        (this.requestPath = undefined),
        (this.requestHeader = undefined),
        (this.cookies = undefined),
        (this.query = undefined),
        (this.body = undefined),
        (this.envInfo = undefined),
        (this.apiType = undefined),
        (this.sort = 0),
        (this.associationExtraction = undefined),
        (this.preScript = undefined),
        (this.postScript = undefined),
        (this.apiResultAssert = undefined),
        (this.sourceDratId = undefined);
    } else {
      (this.parentId = 0),
        (this.projectId = useProjectStore().getProjectId ?? undefined),
        (this.teamId = useTeamStore().getTeamId ?? undefined),
        (this.apiName = '未命名接口'),
        (this.apiNode = ApiNodeType.INTERFACE),
        (this.requestMethod = RequestMethod.GET),
        (this.requestPath = '/api/example'),
        (this.requestHeader = undefined),
        (this.cookies = undefined),
        (this.query = undefined),
        (this.body = new Body()),
        (this.envInfo = new RequestExecuteInfo()),
        (this.apiType = ApiType.HTTP),
        (this.sort = 0),
        (this.associationExtraction = undefined),
        (this.preScript = undefined),
        (this.postScript = undefined),
        (this.mockResponse = new MockResponse()),
        (this.authConfig = new AuthConfig()),
        (this.apiResultAssert = undefined),
        (this.sourceDratId = 0);
    }
  }
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/api/domain/dto/AddApiInterfaceDTO.java
package com.mokatest.platform.demos.api.domain.dto;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import com.mokatest.platform.demos.api.domain.apiEnum.ApiNodeType;
import com.mokatest.platform.demos.api.domain.apiEnum.ApiType;
import com.mokatest.platform.demos.api.domain.apiEnum.RequestMethod;
import com.mokatest.platform.demos.api.domain.requestModel.ApiExtraction;
import com.mokatest.platform.demos.api.domain.requestModel.ApiResponseExample;
import com.mokatest.platform.demos.api.domain.requestModel.AssertParameter;
import com.mokatest.platform.demos.api.domain.requestModel.AuthConfig;
import com.mokatest.platform.demos.api.domain.requestModel.Body;
import com.mokatest.platform.demos.api.domain.requestModel.MockResponse;
import com.mokatest.platform.demos.api.domain.requestModel.RequestExecuteInfo;
import com.mokatest.platform.demos.api.domain.requestModel.RequestParameter;
import com.mokatest.platform.demos.api.domain.requestModel.ResponseSchema;
import com.mokatest.platform.demos.api.domain.requestModel.ScriptItem;
import com.mokatest.platform.demos.api.domain.requestModel.SqlConfig;
import com.mokatest.platform.demos.api.handler.ScriptListTypeHandler;
import lombok.Data;

import java.util.List;

/**
 * @Description: 接口存储传输对象
 * @Author: JingLong
 * @DateTime: 2026/4/3 15:13
 */
@Data
public class AddApiInterfaceDTO {

    private Integer id;

    private Integer parentId;

    /**
     * 所属项目id
     */
    private Integer projectId;

    /**
     * 所属团队id
     */
    private Integer teamId;


    /**
     * 接口名称
     */
    private String apiName;

    /**
     * 节点类型
     */
    private ApiNodeType apiNode;


    /**
     * 请求方式
     */
    private RequestMethod requestMethod;

    /**
     * 请求路径
     */
    private String requestPath;

    /**
     * 请求头
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<RequestParameter> requestHeader;

    /**
     * cookie
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<RequestParameter> cookies;

    /**
     * query参数
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<RequestParameter> query;

    /**
     * body请求体
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private Body body;

    /**
     * 关联的环境数据
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private RequestExecuteInfo envInfo;

    /**
     * 接口类型（HTTP / SQL / TCP / WEBSOCKET）
     */
    private ApiType apiType;

    /**
     * SQL 调试配置（apiType=SQL 时使用）
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private SqlConfig sqlConfig;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 关联提取规则列表
     *
     * 定义从当前接口响应中提取数据的规则，提取结果存入变量池供后续接口使用
     *
     * @see ApiExtraction
     * @since 2026-05-26 类型由 Object 改为 List<ApiExtraction>
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<ApiExtraction> associationExtraction;

    /**
     * 前置脚本列表（按 sort 排序后依次执行）
     *
     * @see ScriptItem
     * @since 2026-05-27 由 Object 改为 List<ScriptItem>
     */
    @TableField(typeHandler = ScriptListTypeHandler.class)
    private List<ScriptItem> preScript;

    /**
     * 后置脚本列表（按 sort 排序后依次执行）
     *
     * @see ScriptItem
     * @since 2026-05-27 由 Object 改为 List<ScriptItem>
     */
    @TableField(typeHandler = ScriptListTypeHandler.class)
    private List<ScriptItem> postScript;

    /**
     * Mock 响应配置
     *
     * @see MockResponse
     * @since 2026-06-17
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private MockResponse mockResponse;

    /**
     * 接口鉴权配置（HTTP 接口调试用）
     *
     * @see AuthConfig
     * @since 2026-08-03
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private AuthConfig authConfig;

    /**
     * Swagger / OpenAPI 响应示例列表
     *
     * @see ApiResponseExample
     * @since 2026-07-03
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<ApiResponseExample> responseExamples;

    /**
     * 响应结构定义（绑定数据模板/内联 schema/覆盖/校验开关）
     *
     * @see ResponseSchema
     * @since 2026-08-06
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private ResponseSchema responseSchema;

    /**
     * 断言
     */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<AssertParameter> apiResultAssert;

    /**
     * 来源id
     */
    private Integer sourceDratId;


}